lib/ansible/modules/crypto/entrust/ecs_certificate.py

```diff
@@ -0,0 +1,906 @@
+        self.cert_days = None
+        self.cert_details = None
+        self.backup_file = None
+
+        self.cert = None
+        self.ecs_client = None
+        if self.path and os.path.exists(self.path):
+            try:
+                self.cert = crypto_utils.load_certificate(self.path, backend='cryptography')
+            except Exception as dummy:
+                self.cert = None
+        # Instantiate the ECS client and then try a no-op connection to verify credentials are valid
+        try:
+            self.ecs_client = ECSClient(
+                entrust_api_user=module.params.get('entrust_api_user'),
+                entrust_api_key=module.params.get('entrust_api_key'),
+                entrust_api_cert=module.params.get('entrust_api_client_cert_path'),
+                entrust_api_cert_key=module.params.get('entrust_api_client_cert_key_path'),
+                entrust_api_specification_path=module.params.get('entrust_api_specification_path')
+            )
+        except SessionConfigurationException as e:
+            module.fail_json(msg='Failed to initialize Entrust Provider: {0}'.format(to_native(e)))
+        try:
+            self.ecs_client.GetAppVersion()
+        except RestOperationException as e:
+            module.fail_json(msg='Please verify credential information. Received exception when testing ECS connection: {0}'.format(to_native(e.message)))
+
+    # Conversion of the fields that go into the 'tracking' parameter of the request object
+    def convert_tracking_params(self, module):
+        body = {}
+        tracking = {}
+        if module.params.get('requester_name'):
+            tracking['requesterName'] = module.params['requester_name']
+        if module.params.get('requester_email'):
+            tracking['requesterEmail'] = module.params['requester_email']
+        if module.params.get('requester_phone'):
+            tracking['requesterPhone'] = module.params['requester_phone']
+        if module.params.get('tracking_info'):
+            tracking['trackingInfo'] = module.params['tracking_info']
+        if(module.params.get('custom_fields')):
+            # Omit custom fields from submitted dict if not present, instead of submitting them with value of 'null'
+            # The ECS API does technically accept null without error, but it complicates debugging user escalations and is unnecessary bandwidth.
+            custom_fields = {}
+            for k, v in module.params['custom_fields'].items():
+                if v is not None:
+                    custom_fields[k] = v
+            tracking['customFields'] = custom_fields
+        if(module.params.get('additional_emails')):
+            tracking['additionalEmails'] = module.params['additional_emails']
+        body['tracking'] = tracking
+        return body
+
+    def convert_cert_subject_params(self, module):
+        body = {}
+        if module.params.get('subject_alt_name') and len(module.params['subject_alt_name']) > 0:
```

Comment:
```suggestion
        if module.params['subject_alt_name']:
```
In Python, the empty string evaluates to `False`.

Reply:
Ah, didn't realize empty lists evaluate to false.